### DISCUSSION TITLE
Problem about deal with new dataset

### DISCUSSION THREAD

**Discussion by jianchaoji** [pyg-team/pytorch_geometric#2721] [Q&A] [answered]
I got a new dataset that consist of two files. One is relation.txt and the other one is feature.txt

The relation file is representing the connection relationship between different nodes like this:
1 0
2 0
3 0
2 1
3 1

And the feature file is the one hot features of each node like this:
1 0 0 1 1 0 0 0
2 0 1 0 0 0 1 0

Can I get some advice about how to transform them to the format like "planetoid" (https://github.com/kimiyoung/planetoid/tree/master/data)

Thank you so much!

**Comment by saiden89** [comment, marked as answer]
Hi,
it would probably be best if you take a look [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/introduction.html) for a well detailed example on how to structure your data for Pytorch Geometric.
In a nutshell, your features are just a `torch.tensor` with shape `(num_nodes, num_features)`. In your case:
```python3
x = torch.tensor([[1, 0, 0, 1, 1, 0, 0, 0], 
                  [2, 0, 1, 0, 0, 0, 1, 0]], dtype=torch.long)
```

As for your `edge_list` (what you call connection relationship), that's another `torch.tensor` of shape `(2, num_edges)`. If you're working with an undirected graph, the index should report both directions. Like so:
```python3
edge_attr = torch.tensor([[1, 0],
                          [0, 1], 
                          [2, 0],
                          [0, 2],                           
                          [3, 0],
                          [0, 3],
                          [2, 1],
                          [1, 2],
                          [3, 1],
                          [1, 3]], dtype=torch.long)
```
That's just the basics, but you will find the Pytorch Geometric is very consistent when it comes to data structures. For a more detailed explanation feel free to browse the linked documentation as well as the worked [examples](https://github.com/rusty1s/pytorch_geometric/tree/master/examples) in this GitHub repository. There are also some Colab [notebooks](https://pytorch-geometric.readthedocs.io/en/latest/notes/colabs.html) if you prefer.

**Reply by rusty1s** [reply]
To follow-up on @saiden89, you can read your txt files in via standard Python file reading, or more advanced techniques such as `pandas.read_csv`. This will get you the aforementioned tensors as numpy arrays.

**Reply by jianchaoji** [reply]
I will have a try. Thank you!
